src/orchestrator.ts

```diff
@@ -133,21 +133,13 @@ export class Orchestrator {
                     return;
                 }
 
+                const completionIndex = partialResult.completionIndex;
+
                 // The first time a task is marked as complete, the history event that finally marked the task as completed
                 // should not yet have been played by the Durable Task framework, resulting in isReplaying being false.
                 // On replays, the event will have already been processed by the framework, and IsPlayed will be marked as true.
-                if (state[partialResult.completionIndex] !== undefined) {
-                    context.df.isReplaying = state[partialResult.completionIndex].IsPlayed;
-                }
-
-                if (TaskFilter.isFailedTask(partialResult)) {

```

Comment:
I'm trying to think if there are many major implications of moving this. We are essentially moving it past the case where the customer `returns` a non-yielded completed task, which shouldn't affect this code path, as well as after we update the replay state. 

I am pretty sure this is all safe, just wanted to highlight any potential changes this would cause.

Reply:
+1, I was also quite worried about this change but I agree that this appears safe. Another way to think about it, which helps in building confidence about this change, is that we've moved it right next to where we feed a successful value to the generator via `gen.next`. It makes sense that `gen.next` would be right next to `gen.throw` and since `gen.next` works correctly, so should this case :)